src/main/java/com/yelp/nrtsearch/server/luceneserver/search/SearchRequestProcessor.java

```diff
@@ -148,6 +152,14 @@ public static SearchContext buildContextForRequest(
         getRescorers(indexState, searcherAndTaxonomy.searcher, searchRequest));
     contextBuilder.setSharedDocContext(new DefaultSharedDocContext());
 
+    Highlight highlight = searchRequest.getHighlight();
+    if (!highlight.getFieldsList().isEmpty()) {
+      verifyHighlights(indexState, highlight);
+      HighlightContext highlightContext =
+          new HighlightContext(indexState, searcherAndTaxonomy, query, highlight);
+      contextBuilder.setHighlightContext(highlightContext);
```

Comment:
Another way to do this would be to create a fetch task instance, and store that in the `SearchContext` instead of the `HighlightContext`. This would simplify the interface between the search and highlights modules, and having a stateful task would make future extensions (additional highlighter types, different types per field, etc) more straightforward.